### PR TITLE
.github: add e2e job

### PR DESCRIPTION
**Pull request by pierre**
Ruby (core) tests are running again.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,3 +8,161 @@ on:
...
+        if: ${{ matrix.database == 'postgresql' }}
+        run: |
+          curl https://raw.githubusercontent.com/killbill/killbill-avatax-plugin/master/src/main/resources/ddl.sql | psql -h 127.0.0.1 -U postgres -p 5432 -d killbill
+          curl https://raw.githubusercontent.com/killbill/killbill-payment-test-plugin/master/src/main/resources/ddl.sql | psql -h 127.0.0.1 -U postgres -p 5432 -d killbill
+          curl https://raw.githubusercontent.com/killbill/killbill-email-notifications-plugin/master/src/main/resources/ddl.sql | psql -h 127.0.0.1 -U postgres -p 5432 -d killbill
+      - name: Start Kill Bill
+        timeout-minutes: 4
+        run: |
+          mkdir -p /tmp/test-results
+          nohup ./bin/start-server -s > /tmp/test-results/killbill.log 2>&1 &
```
Comment (sbrossie) [inline .github/workflows/ci.yml:109]: Should we either modify our default `profiles/killbill/src/main/resources/killbill-server.properties` to have the `org.killbill.security.shiroNbHashIterations=1` or create a special one for the test?

Comment (sbrossie) [reply .github/workflows/ci.yml:109]: Also, I am confused how it works for the `postgresql' `matrix.database` - wouldn't we need to modify the `org.killbill.dao.url` property?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -8,3 +8,161 @@ on:
...
+    strategy:
+      matrix:
+        include:
+          - ruby-version: '2.4.2'
+            ref: refs/heads/master
+            database-adapter: 'mysql2'
+            database-user: 'root'
+            database-password: 'root'
+            database-port: '3306'
+            database: 'mysql'
```
Comment (sbrossie) [inline .github/workflows/ci.yml:30]: Seems like the job supports different values for `database` but all properties here are targeted for `mysql` - Am i missing something?

**Review by sbrossie — APPROVED** [review]
👏

File: .github/workflows/ci.yml (hunk 1 above)
Comment (pierre) [reply .github/workflows/ci.yml:109]: > Should we either modify our default `profiles/killbill/src/main/resources/killbill-server.properties` to have the `org.killbill.security.shiroNbHashIterations=1` or create a special one for the test?

👍 for changing the default. Maybe in `work-for-release-0.23.x`?

> Also, I am confused how it works for the `postgresql` `matrix.database` - wouldn't we need to modify the `org.killbill.dao.url` property?

Yup, the PostgreSQL setup is only 80% done. I wanted to see how stable the MySQL one is for a few weeks first, before also enabling PostgreSQL.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (pierre) [reply .github/workflows/ci.yml:30]: PostgreSQL support is for future work 😄 

We have it enabled on the other repo (https://github.com/killbill/killbill-integration-tests/actions/runs/1863918667), but I just wanted to see how stable the job is here, as the setup is a tad different.

**Pull request merged**